core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java

```diff
@@ -309,7 +310,7 @@ static QueryStateMachine beginWithTicker(
             session = session.withExchangeEncryption(serializeAesEncryptionKey(createRandomAesEncryptionKey()));
         }
 
-        if (!queryType.map(SELECT::equals).orElse(false)) {
+        if (!queryType.map(SELECT::equals).orElse(false) || !isSpoolingProtocolEnabled(session)) {
```

Comment:
This is a bit weird to have same information expressed in a session object by two means...

Not terrible though.